crates/fluvio-controlplane-metadata/src/managed_connector/spec.rs

```diff
@@ -0,0 +1,29 @@
+#![allow(clippy::assign_op_pattern)]
+
+use dataplane::core::{Encoder, Decoder};
+
+#[derive(Encoder, Decoder, Default, Debug, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "use_serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "camelCase")
+)]
+pub struct ManagedConnectorSpec {
+    pub name: String,
+    pub config: ManagedConnectorConfig,
+}
+
+#[derive(Encoder, Decoder, Default, Debug, Clone, PartialEq)]
+#[cfg_attr(
+    feature = "use_serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "camelCase")
+)]
+pub struct ManagedConnectorConfig {
+    #[cfg_attr(feature = "use_serde", serde(rename = "type"))]
+    pub type_: String, // syslog, github star, slack
```

Comment:
Since "type" in often used to refer the programming language construct, especially in Rust, this may lead to some confusion and awkwardness when communicating. I suggest using "package", as it will refer to the connector packages that we will maintain either in our registry or on GH.